tests/test_logic.py

```diff
@@ -494,3 +494,143 @@ def test_isclose(dtype, rtol, atol):
+    yield (e0, e0.copy(), True, True)
+
+    #
+    yield (e1, e1.copy(), None, True)
+    yield (e1, e1.copy(), False, True)
+    yield (e1, e1.copy(), True, True)
+
+    # Non-nanable – those cannot hold nans
+    a12 = dpnp.array([1, 2])
+    a12b = a12.copy()
+    a123 = dpnp.array([1, 2, 3])
+    a13 = dpnp.array([1, 3])
+    a34 = dpnp.array([3, 4])
+
+    yield (a12, a12b, None, True)
+    yield (a12, a12, None, True)
+    yield (a12, a123, None, False)
+    yield (a12, a34, None, False)
+    yield (a12, a13, None, False)
+
+    # Non-float dtype - equal_nan should have no effect,
+    yield (a123, a123, None, True)
+    yield (a123, a123, False, True)
+    yield (a123, a123, True, True)
+    yield (a123, a123.copy(), None, True)
+    yield (a123, a123.copy(), False, True)
+    yield (a123, a123.copy(), True, True)
+    yield (a123.astype("f4"), a123.astype("f4"), None, True)
+    yield (a123.astype("f4"), a123.astype("f4"), False, True)
+    yield (a123.astype("f4"), a123.astype("f4"), True, True)
+
+    # these can hold None
+    b1 = dpnp.array([1, 2, dpnp.nan])
+    b2 = dpnp.array([1, dpnp.nan, 2])
+    b3 = dpnp.array([1, 2, dpnp.inf])
+    b4 = dpnp.array(dpnp.nan)
+
+    # instances are the same
+    yield (b1, b1, None, False)
+    yield (b1, b1, False, False)
+    yield (b1, b1, True, True)
+
+    # equal but not same instance
+    yield (b1, b1.copy(), None, False)
+    yield (b1, b1.copy(), False, False)
+    yield (b1, b1.copy(), True, True)
+
+    # same once stripped of Nan
+    yield (b1, b2, None, False)
+    yield (b1, b2, False, False)
+    yield (b1, b2, True, False)
+
+    # nan's not conflated with inf's
+    yield (b1, b3, None, False)
+    yield (b1, b3, False, False)
+    yield (b1, b3, True, False)
+
+    # all Nan
+    yield (b4, b4, None, False)
+    yield (b4, b4, False, False)
+    yield (b4, b4, True, True)
+    yield (b4, b4.copy(), None, False)
+    yield (b4, b4.copy(), False, False)
+    yield (b4, b4.copy(), True, True)
+
+    # Multi-dimensional array
+    md1 = dpnp.array([[0, 1], [dpnp.nan, 1]])
+
+    yield (md1, md1, None, False)
+    yield (md1, md1, False, False)
+    yield (md1, md1, True, True)
+    yield (md1, md1.copy(), None, False)
+    yield (md1, md1.copy(), False, False)
+    yield (md1, md1.copy(), True, True)
+    # both complexes are nan+nan.j but the same instance
+    cplx1, cplx2 = [dpnp.array([dpnp.nan + dpnp.nan * 1j])] * 2
+
+    # only real or img are nan.
+    cplx3 = dpnp.array(1 + dpnp.nan * 1j)
+    cplx4 = dpnp.array(dpnp.nan + 1j)
+
+    # Complex values
+    yield (cplx1, cplx2, None, False)
+    yield (cplx1, cplx2, False, False)
+    yield (cplx1, cplx2, True, True)
+
+    # Complex values, 1+nan, nan+1j
+    yield (cplx3, cplx4, None, False)
+    yield (cplx3, cplx4, False, False)
+    yield (cplx3, cplx4, True, True)
+
+
+class TestArrayComparisons:
+    @pytest.mark.parametrize(
+        "bx,by,equal_nan,expected", _test_array_equal_parametrizations()
```

Comment:
I don't think it is a good idea just to copy paste tests from numpy.
Numpy doesn't have an array to compare, but dpnp usually compares again numpy result. And it isn't the case here.
In general, we can reuse the test scenario which numpy has, but it'd be great to slightly modify them to the way we implement tests in dpnp.